src/Mailbox.php

```diff
@@ -41,6 +42,20 @@ public function getName(): string
         return $this->name;
     }
 
+    public function setName(string $name): bool
+    {
+        $oldFullName = $this->getFullEncodedName();
+        $newFullName = str_replace($this->name, $name, $oldFullName);
```

Comment:
This may replace also parent folder's names: without using a regex, we can just take the last position of `$this->name` in `$oldFullName` with [`strrpos`](https://www.php.net/manual/en/function.strrpos.php), strip it and append the new one